rotkehlchen/api/v1/schemas.py

```diff
@@ -3031,3 +3031,134 @@ class EthStakingHistoryStatsDetails(EthStakingHistoryStats, AsyncIgnoreCacheQuer
 
 class SkippedExternalEventsExportSchema(Schema):
     filepath = FileField(required=True, allowed_extensions=['.csv'])
+
+
+class ExportHistoryEventSchema(TimestampRangeSchema, DBOrderBySchema):
+    """Schema for quering history events"""
+    directory_path = DirectoryField(required=True)
+    exclude_ignored_assets = fields.Boolean(load_default=True)
+    event_identifiers = DelimitedOrNormalList(fields.String(), load_default=None)
+    event_types = DelimitedOrNormalList(
+        SerializableEnumField(enum_class=HistoryEventType),
+        load_default=None,
+    )
+    event_subtypes = DelimitedOrNormalList(
+        SerializableEnumField(enum_class=HistoryEventSubType),
+        load_default=None,
+    )
+    location = SerializableEnumField(Location, load_default=None)
+    location_labels = DelimitedOrNormalList(fields.String(), load_default=None)
+    asset = AssetField(expected_type=CryptoAsset, load_default=None)
+    entry_types = IncludeExcludeListField(
+        SerializableEnumField(enum_class=HistoryBaseEntryType),
+        load_default=None,
+    )
+
+    # EvmEvent only
+    tx_hashes = DelimitedOrNormalList(EVMTransactionHashField(), load_default=None)
+    counterparties = DelimitedOrNormalList(fields.String(), load_default=None)
+    products = DelimitedOrNormalList(SerializableEnumField(enum_class=EvmProduct), load_default=None)  # noqa: E501
+    addresses = DelimitedOrNormalList(EvmAddressField(), load_default=None)
+
+    # EthStakingEvent only
+    validator_indices = DelimitedOrNormalList(fields.Integer(), load_default=None)
+
+    @validates_schema
+    def validate_history_event_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        valid_ordering_attr = {None, 'timestamp'}
+        if (
+            data['order_by_attributes'] is not None and
+            not set(data['order_by_attributes']).issubset(valid_ordering_attr)
+        ):
+            error_msg = (
+                f'order_by_attributes for history event data can not be '
+                f'{",".join(set(data["order_by_attributes"]) - valid_ordering_attr)}'
+            )
+            raise ValidationError(
+                message=error_msg,
+                field_name='order_by_attributes',
+            )
+
+    @post_load
+    def make_history_event_filter(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> dict[str, Any]:
+        should_query_eth_staking_event = data['validator_indices'] is not None
+        should_query_evm_event = any(data[x] is not None for x in ('products', 'counterparties', 'tx_hashes', 'addresses'))  # noqa: E501
+        counterparties = data['counterparties']
+        entry_types = data['entry_types']
+        if counterparties is not None and CPT_ETH2 in counterparties:
```

Comment:
this function is also a copy paste with the exception of the limit/offset. If you wanted to remove this fields only you could have made an abstraction that took the common arguments as argument. 

Have you coordinated with the frontend how are they going to call this endpoint. If they don't pass the limit/offset is enough with using the same schema. What I would do is use the same methods in both but have 


```python
class HistoryEventSchema(TimestampRangeSchema, DBOrderBySchema):
    # the one you are adding in this PR

class PaginatedHistoryEventSchema(DBPaginationSchema,  HistoryEventSchema):
    # the existing one
```

and in this one instead of doing

```python
            'limit': data['limit'],
            'offset': data['offset'],
```

do 

```python
            'limit': data.get('limit'),
            'offset': data.get('offset'),
```

`.get` in python returns `None` when the value doesn't exist. If you can find a better way of doing this go with it but don't duplicate ever code. If you need to duplicate code something is bad designed

Reply:
I did a work around to extend the previous class without breaking any logic